src/components/spotlight/SpotlightModalForStep.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { useEffect } from 'react';
+import { useIntl } from 'react-intl';
+import { TooltipRenderProps } from 'react-joyride';
+import { Button, ButtonGroup, ButtonVariety } from '../buttons';
+import { SpotlightModal } from './SpotlightModal';
+import { SpotlightProps, SpotlightStep } from './SpotlightTypes';
+
+export type SpotlightModalForStepProps = TooltipRenderProps &
+  Pick<SpotlightProps, 'activeTargetClassName' | 'stepLink' | 'stepXofYLabel'> & {
+    /**
+     * The react-joyride props for this Spotlight step- see https://docs.react-joyride.com/step
+     */
+    step: SpotlightStep;
+  };
+
+export function SpotlightModalForStep(props: Readonly<SpotlightModalForStepProps>) {
+  const {
+    activeTargetClassName = 'active',
+    backProps,
+    closeProps,
+    continuous,
```

Comment:
isContinuous?

Reply:
I don't think so, we are doing this in many places to make sure we keep a consistent API of Echoes. For example if you look at the Select where we use mantine we have plenty of props renamed to suit us.

Doing that also allow us to provide proper documentation for these props in our TSDoc if needed.
It also kind of forces us to be exhaustive with the API we expose and it make sure we don't allow un-expected behaviors that could be introduced without us noticing through new props added with a react-joyride upgrade.